summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala

```diff
@@ -21,43 +21,65 @@ import scala.reflect.ClassTag
 case class StormTopologyBuilder(options: Map[String, Options], jobId: JobId, stormDag: Dag[Storm]) {
   @transient private val logger = LoggerFactory.getLogger(classOf[StormTopologyBuilder])
 
-  def build(): StormTopology = {
-    val topologyBuilder = new TopologyBuilder
-    stormDag.nodes.foreach {
-      case summerNode: SummerNode[Storm] => scheduleSummerBolt(summerNode, topologyBuilder)
-      case flatMapNode: FlatMapNode[Storm] => scheduleFlatMapper(flatMapNode, topologyBuilder)
-      case sourceNode: SourceNode[Storm] => scheduleSpout(sourceNode, topologyBuilder)
-    }
-    topologyBuilder.createTopology()
-  }
+  def build: StormTopology = registerEdges(registerNodes).build(jobId)
 
-  /**
-   * Set storm to tick our nodes every second to clean up finished futures
-   */
-  private def tickConfig = {
-    val boltConfig = new BacktypeStormConfig
-    boltConfig.put(BacktypeStormConfig.TOPOLOGY_TICK_TUPLE_FREQ_SECS, java.lang.Integer.valueOf(1))
-    boltConfig
-  }
+  private def registerNodes: Topology =
+    stormDag.nodes.foldLeft(Topology.EMPTY) {
+      case (topology, node: SummerNode[Storm]) =>
+        topology.withBolt(getNodeName(node), registerSummerBolt(node))._2
+      case (topology, node: FlatMapNode[Storm]) =>
+        topology.withBolt(getNodeName(node), FlatMapBoltProvider(this, node).apply)._2
+      case (topology, node: SourceNode[Storm]) =>
+        topology.withSpout(getNodeName(node), SpoutProvider(this, node).apply)._2
+    }
 
-  private def scheduleFlatMapper(node: FlatMapNode[Storm], topologyBuilder: TopologyBuilder) = {
-    val bolt: BaseBolt[Any, Any] = FlatMapBoltProvider(this, node).apply
+  private def registerEdges(topologyWithNodes: Topology): Topology =
+    stormDag.nodes.foldLeft(topologyWithNodes) {
+      case (topology, node: SummerNode[Storm]) =>
+        val edgeType = EdgeType.AggregatedKeyValues[Any, Any](getSummerKeyValueShards(node))
+        registerIncomingEdges(topology, node, edgeType)
+      case (topology, node: FlatMapNode[Storm]) =>
+        registerIncomingEdges(topology, node, flatMapEdgeType(node))
+      case (topology, node: SourceNode[Storm]) => topology
+        // ignore, no incoming edges into sources
```

Comment:
I can check there is no incoming edges here, but actually it's impossible because of types. So I believe ignore is what we need.